docs/notary.md

```diff
@@ -0,0 +1,531 @@
+   transaction field). Use the following rules to construct the list:
+   * First signer is the one who pays transaction fees.
+   * Each signer is either multisignature or standard signature or a contract
+     signer.
+   * Multisignature and signature signers can't be combined.
+   * Contract signer can be combined with any other signer.
+   * Maximum number of multisignature signers is 1.
+   * Maximum number of signature or contract signers is unlimited.
+
+   Include Notary native contract in the list of signers with the following
+   constraints:
+   * Notary signer hash is the hash of native Notary contract that can be fetched
+     from
+     [func (*Client) GetNativeContractHash](https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.97.2/pkg/rpc/client#Client.GetNativeContractHash).
+   * Notary signer must have `None` scope.
+   * Notary signer shouldn't be placed at the beginning of the signer list,
+     because Notary contract does not pay main transaction fees. Other positions
+     in the signer list are available for Notary signer.
+2. Construct script for the main transaction (that will be `Script` transaction
+   field) and calculate system fee using regular rules (that will be `SystemFee`
+   transaction field). Probably, you'll perform one of these actions:
+   1. If the script is a contract method call, use `invokefunction` RPC API
+      [func (*Client) InvokeFunction](https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.97.2/pkg/rpc/client#Client.InvokeFunction)
+      and fetch script and gas consumed from the result.
+   2. If the script is more complicated than just a contract method call,
+      construct the script manually and use `invokescript` RPC API
+      [func (*Client) InvokeScript](https://pkg.go.dev/github.com/nspcc-dev/neo-go@v0.97.2/pkg/rpc/client#Client.InvokeScript)
+      to fetch gas consumed from the result.
+   3. Or just construct the script and set system fee manually.
+3. Calculate the height main transaction is valid until (that will be
+   `ValidUntilBlock` transaction field). Consider the following rules for `VUB`
+   value estimation:
+      * `VUB` value must not be lower than current chain height.
+      * The whole notary request (including fallback transaction) is valid until
+        the same `VUB` height.
+      * `VUB` value must be lower than notary deposit expiration height. This
+        condition guarantees that deposit won't be withdrawn before notary
+        service payment.
+4. Construct the list of main transaction attributes (that will be `Attributes`
+   transaction field). The list must include `NotaryAssisted` attribute with
+   `NKeys` equals to the sum number of keys to be collected excluding notary and
+   other contract-based witnesses. For m out of n multisignature request
+   `NKeys = n`. For multiple standard signature request signers `NKeys` equals to
+   the standard signature signers count.
+5. Construct the list of accounts to calculate network fee for the transaction
```

Comment:
Maybe it's worth noting that it's `wallet.Account` accounts we're talking about here.

Reply:
We don't have godoc for that, so I just added package and structure names.